python/image-content-search/stack/cdk.py

```diff
@@ -0,0 +1,369 @@
+import yaml
+
+from aws_cdk import (
+    aws_s3_notifications as _s3notification,
+    aws_lambda_event_sources as _lambda_event_source,
+    aws_s3 as _s3,
+    aws_cognito as _cognito,
+    aws_sqs as _sqs,
+    aws_iam as _iam,
+    aws_events as _events,
+    aws_events_targets as _event_targets,
+    aws_rds as _rds,
+    aws_secretsmanager as _secrets_manager,
+    custom_resources as _custom_resources,
+    core
+)
+
+from aws_cdk.aws_apigateway import (
+    RestApi, 
+    LambdaIntegration,
+    CfnAuthorizer,
+    AuthorizationType,
+    MockIntegration,
+    PassthroughBehavior
+)
+
+from aws_cdk.aws_lambda import (
+    Code, 
+    Function,
+    Runtime
+)
+
+from aws_cdk.custom_resources import (
+    AwsCustomResource,
+    AwsCustomResourcePolicy,
+    AwsSdkCall,
+    PhysicalResourceId,
+    Provider
+)
+
+class ImageContentSearchStack(core.Stack):
+
+    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 96 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 159-338 into a separate function.